packages/shorebird_cli/lib/src/executables/aot_tools.dart

```diff
@@ -98,21 +104,78 @@ class AotTools {
     // distributed as an executable.
     final extension = p.extension(artifactPath);
     if (extension != '.dill' && extension != '.dart') {
-      return process.run(
+      return runFn(
         artifactPath,
         command,
         workingDirectory: workingDirectory,
       );
     }
 
     // local engine versions use .dart and we distribute aot-tools as a .dill
-    return process.run(
+    return runFn(
       shorebirdEnv.dartBinaryFile.path,
       ['run', artifactPath, ...command],
       workingDirectory: workingDirectory,
     );
   }
 
+  Future<ShorebirdProcessResult> _exec(
```

Comment:
The naming and nesting is a bit confusing imo. We have `prepareAndRun`, `exec`, `spawn` and it's not clear what the difference is between them imo.